File: Programing_Basics_with_JS/forLoopMoreExercises/01.BackToThePast.js
function backToThePast(input) {
    let money = Number(input[0]);
    let year = Number(input[1]);

    let neededMoney = 0;
    let age = 18;

    for (let i = 1800; i <= year; i++) {
        if (i % 2 == 0) {
            neededMoney += 12000;
            age ++;
        } else {
            neededMoney += 12000 + 50 * age;
            age++;
        }
    }

    let moneyDiff = Math.abs(money - neededMoney);

    if (money >= neededMoney) {
        console.log(`Yes! He will live a carefree life and will have ${moneyDiff.toFixed(2)} dollars left.`);
    } else {
        console.log(`He will need ${moneyDiff.toFixed(2)} dollars to survive.`);
    }
}

backToThePast(['50000', '1802'])
backToThePast(['100000.15', '1808'])

File: Programming_Basics_with_Python/for_loop_more_exercises/01._back_to_the_past.py
legacy_money = float(input())
year_to_live = int(input())

money_cost_per_year = 12000
ivan_years = 18
back_to_past = 1800

for year in range(back_to_past, year_to_live + 1):
    if year % 2 == 0:
        legacy_money -= money_cost_per_year
    else:
        legacy_money -= money_cost_per_year + (50 * ivan_years)
    ivan_years += 1

if legacy_money >= 0:
    print(f'Yes! He will live a carefree life and will have {legacy_money:.2f} dollars left.')
else:
    print(f'He will need {abs(legacy_money):.2f} dollars to survive.')